compiler/include/baseAST.h

```diff
@@ -288,6 +288,9 @@ static inline bool isSymbol(const BaseAST* a)
 static inline bool isType(const BaseAST* a)
 { return a && isType(a->astTag); }
 
+static inline bool isLcnSymbol(const BaseAST* a)
```

Comment:
IIRC, the rationale for using tags and our own is*() routines for all AST nodes is that the C++ RTT mechanisms were considered heavier-weight/slower by developers with more C++ experience than I had.  That decision was made a decade ago, though, so I'm not confident whether or not it's still true.  However, I wouldn't move away from the current approach without checking that.  

At any rate, adding that as a TODO to this specific patch seems misplaced -- a pivotal task about cleanup seems more appropriate given that it's a broad change unrelated to LcnSymbol specifically.